### DISCUSSION TITLE
How to concisely use <Controller> with semantic-ui-react

### DISCUSSION THREAD

**Discussion by noomly** [react-hook-form/react-hook-form#6097] [Q&A] [answered]
I am having an error when trying to idiomatically use the `Controller` component to render a `Select` from semantic-ui-react like this:

```typescript
<Controller
    control={control}
    name="name"
    rules={{ required: true }}
    render={({ field }) => (
        <Select
            {...field}
            placeholder="Select a name"
            options={names}
        />
    )}
/>
```

The error being: `Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?`.

I am aware that it is because `Select` doesn't expose the `ref` prop. My question is what is the cleanest way to write an equivalent that works using a similar syntax (`{...field}`) to keep the code succinct?

EDIT: forgot to mention I'm using react-hook-form version 7.5

**Comment by bluebill1049** [comment]
you can omit the `ref`, however if you omit the `ref` then the focus is probably not going to work.

**Reply by noomly** [reply]
Thanks for your answer but I was looking for a solution that would cover the same level of functionality. See my own answer below for such a solution.

**Comment by noomly** [comment, marked as answer]
In case someone stumble upon the same issue, I found that semantic-ui-react has a component `Ref` made exactly for this kind of case. The solution would be:

```typescript
<Controller
    control={control}
    name="name"
    rules={{ required: true }}
    render={({ field }) => (
        <Ref innerRef={field.ref}>
            <Select
                {...field}
                ref={null}
                placeholder="Select a name"
                options={names}
            />
        </Ref>
    )}
/>
```
